suite-native/module-authorize-device/src/components/connect/PinOnKeypad.tsx

```diff
@@ -74,14 +79,28 @@ export const PinOnKeypad = ({ variant, onSuccess }: PinOnKeypadProps) => {
         },
     });
 
+    const [isImageDisplayed, setIsImageDisplayed] = useState(true);
+
+    useEffect(() => {
+        const { height, scale, fontScale } = windowDimensions;
+
+        // If user has too small device or uses accessibility settings for screen content enlargement,
+        // the image is hidden to make sure the the content fits in the screen.
+        if (height < 720 || fontScale > 1 || scale > 3) {
+            setIsImageDisplayed(false);
+        } else {
+            setIsImageDisplayed(true);
+        }
```

Comment:
I'd prefer to show what conditions need to be met to **show** the image and then simplify like this:
```suggestion
        setIsImageDisplayed(height >= 720 && fontScale <= 1 && scale <= 3);
```